### ISSUE TITLE
Failed to post-process. Returned log from SickBeard was not as expected

### ISSUE THREAD

**Issue by dltyih** [clinton-hall/nzbToMedia#909]
I had to reinstall on my QNAP TVS-871 (Firmware 4.2) with SickBeard (TVRage fork, master / commit: 4da36fb1c4fde840cb5ddae0da65fdb56a31c2d0), SABnzbd+ (0.7.20), git 2.1.0 and Python 2.7.3 from scratch (unfortunately the config backups were lost as well). The SR, SAB+ and git are all from Clinton Hall's packages. Python 2.7.3 from the QNAP store.

I have patched the SABnzbd+ to not look for SSL, and set up following directions in the wiki. However, I keep getting PP error message from SABnzbd+ telling me that the returned log is not what SB expected. I had verified that the show was snatched by SB, downloaded by SAB+ but not renamed. I checked several times but could not for the life of me identify where the problem was. Can someone help?

Here is the log for a failed PP:

[SB:nzbToMedia Error Log.txt](https://github.com/clinton-hall/nzbToMedia/files/57060/SB.nzbToMedia.Error.Log.txt)

I would be very grateful for any help / pointers. 

Many thanks!

**Comment by noharm1010** [comment]
same error here, almost same configuration.
Did a complete re-installation already, did not succeed with that.
I think it started after one update a few days ago.

**Comment by madmanx2** [comment]
I'm getting a similar error with Sickrage (BRANCH: (master) / COMMIT: (4da36fb1c4fde840cb5ddae0da65fdb56a31c2d0)  and nzbToMedia MAIN: nzbToMedia Version:1cd64053b0da1d351bb6af392e1ebb904c6b6ce2 Branch:master (Windows 8)).   Tried re-starting both Sickrage and nzbget and manually post processing from nzbget, still results in the log.

Link to Log from point of manual request to re-postprocess the file, executed from within nzbget:

http://pastebin.com/YWdB5ZDX

**Comment by fuuss** [comment]
How can I switch to the nightly branch on my qnap  ?

**Comment by madmanx2** [comment]
@fuuss 
I followed these instructions from the sickrage forums:

https://sickrage.tv/forums/forum/help-support/nzbtomedia-support/31073-help-failed-to-post-process-returned-log-from-sickbeard-was-not-as-expected/page2

I found that when I went in to contrib/nzbToMedia a git pull returned:

Code:
mdpauley@nuc-server:/opt/sickbeard/contrib/nzbToMedia$ git pull
You are not currently on a branch. Please specify which
branch you want to merge with. See git-pull(1) for details.

    git pull <remote> <branch>
I did:

Code:
mdpauley@nuc-server:/opt/sickbeard/contrib/nzbToMedia$ git checkout nightly
Switched to branch 'nightly'
Your branch is up-to-date with 'origin/nightly'.
And now post processing works

**Comment by fuuss** [comment]
I get this error git checkout nightly
fatal: Not a git repository (or any parent up to mount point /share/MD0_DATA)
Stopping at filesystem boundary (GIT_DISCOVERY_ACROSS_FILESYSTEM not set).
[/share/MD0_DATA/Download/sabnzbd/scripts] #

**Comment by clinton-hall** [comment]
@fuuss what git qpkg do you have installed?

I am going to try and merge all changes up to master to get this working...

**Comment by dltyih** [comment]
Thank you, all. I checked out the link @madmanx2 and found someone only changed the fork in autoProcessMedia.cfg from auto to sickrage. I did the same and it seems to have worked for me. I have not thoroughly tested it yet but at least a few test downloads have gone through.

**Comment by fuuss** [comment]
I never installed a qpkg for nzbtomedia I just downloaded it and put in in the scripts folder of sabnzbd . Is there a qpkg for nzbtomedia ?

**Comment by clinton-hall** [comment]
@fuuss 
you should download via git.... this will allow you to use the git commands.
Otherwise, you should just update to the latest version of nzbToMedia by downloading the latest and replacing the exisitng files.

**Comment by fuuss** [comment]
Is there some tutorial  how to  download install via git ?
Do I need to delete this first ?
 Can I install in the sickbeard script folder or do I have to create another one ?
Thank you for the help

**Comment by clinton-hall** [comment]
assuming there are no other scripts in scripts directory... then via ssh:

```
cd /share/MD0_DATA/Download/sabnzbd/
cp -rf scripts scriptsold
git clone http://github.com/clinton-hall/nzbToMedia.git scripts
cp scriptsold/autoProcessMedia.cfg scripts
```

if everything works ok, you can then delete scriptsold
```
cd /share/MD0_DATA/Download/sabnzbd/
rm -rf scriptsold
```

**Comment by JoLeander899** [comment]
Failed to post-process - Returned log from SickBeard was not as expected.!

This post is informational in the hopes it helps others -- 
I have this same scenario whether running on linux or Windows  -- all programs are up-to-date

I receive this error in any and all downloads that are in my que on SABnzbd when I have to pause downloading for any length of time (old hardware -- can't download and watch at the same time) -- when my downloading continues after pause the downloads are successful but my nzbtoSickbeard.py fails.  (Manual postprocessing of the downloaded files in SickRage is successful).  Any downloads added to the SABnzbd que after the ones that were "hanging in line" during the pause do not have any problems automatically postprocessing.  No restarts or reboot necessary to correct the problem.  

[error1.txt](https://github.com/clinton-hall/nzbToMedia/files/66823/error1.txt)

Short version of error --
nzbToMedia Version:3e4861e87c543a4d2debd8997feb16bda3ef5339 Branch:master (Linux 3.16.
[POSTPROCESS]::SICKBEARD: SUCCESS: The download succeeded, sending a post-process request
 [ERROR]::MAIN: A problem was reported in the /home/susan/.nzbtomedia/nzbToSickBeard.py script.
SickBeard: Failed to post-process - Returned log from SickBeard was not as expected.!

**Comment by clinton-hall** [comment]
@JoLeander899 
Please enable debug logging
```
[General]
    log_debug = 1
```
and then post debug logs when this happens again.

I can't see how the fact that the download was paused at any time could influence this behaviour. there are no inherited environment variables etc or argumentspassed through from SABnzbd that would be influenced here.

best guess, if you need to pause downloading while watching, is that while you are watching TV your sickrage is becoming bogged-down and not returning the right output when we do the auto-detection...

**Comment by Calimerorulez** [comment]
I'm seeing the same problem, after updating to the latest version of Sickrage. I'm currently using the nightlies as my last resort :-)

Log is as follows:

2015-12-30 17:02:48 INFO    ::MAIN: Checking database structure...
2015-12-30 17:02:48 DEBUG   ::MAIN: Checking Initial Schema database upgrade
2015-12-30 17:02:48 DEBUG   ::MAIN: InitialSchema upgrade not required
2015-12-30 17:02:48 INFO    ::MAIN: Checking if source needs an update
2015-12-30 17:02:49 DEBUG   ::MAIN: cur_commit = None % (newest_commit)= 3e4861e87c543a4d2debd8997feb16bda3ef5339, num_commits_behind = 0
2015-12-30 17:02:49 ERROR   ::MAIN: Unknown current version number, don't know if we should update or not
2015-12-30 17:02:49 INFO    ::MAIN: nzbToMedia Version:10.13 Branch:master (FreeBSD 9.1-RELEASE)
2015-12-30 17:02:50 WARNING ::MAIN: Failed to locate 7zip. Transcosing of disk images and extraction of .7z files will not be possible!
2015-12-30 17:02:50 WARNING ::MAIN: Failed to locate ffmpeg. Transcoding disabled!
2015-12-30 17:02:50 WARNING ::MAIN: Install ffmpeg with x264 support to enable this feature  ...
2015-12-30 17:02:50 WARNING ::MAIN: Failed to locate ffprobe. Video corruption detection disabled!
2015-12-30 17:02:50 WARNING ::MAIN: Install ffmpeg with x264 support to enable this feature  ...
2015-12-30 17:02:50 INFO    ::MAIN: #########################################################
2015-12-30 17:02:50 INFO    ::MAIN: ## ..::[nzbToMedia.pyc]::.. ##
2015-12-30 17:02:50 INFO    ::MAIN: #########################################################
2015-12-30 17:02:50 DEBUG   ::MAIN: Options passed into nzbToMedia: ['/mnt/media/nzbget/postproc/nzbToSickBeard.py']
2015-12-30 17:02:50 INFO    ::MAIN: Script triggered from NZBGet Version 16.4.
2015-12-30 17:02:50 DEBUG   ::MAIN: Adding NZB download info for directory /mnt/media/downloads/tv/Homeland.S05E12.A.False.Glimmer.1080p.WEB-DL.DD5.1.H.264-NTb to database
2015-12-30 17:02:50 INFO    ::MAIN: Auto-detected SECTION:SickBeard
2015-12-30 17:02:50 DEBUG   ::MAIN: Checking for archives to extract in directory: /mnt/media/downloads/tv/Homeland.S05E12.A.False.Glimmer.1080p.WEB-DL.DD5.1.H.264-NTb
2015-12-30 17:02:50 INFO    ::MAIN: Calling SickBeard:tv to post-process:Homeland.S05E12.A.False.Glimmer.1080p.WEB-DL.DD5.1.H.264-NTb
2015-12-30 17:02:52 INFO    ::MAIN: Attempting to auto-detect tv fork
2015-12-30 17:02:52 INFO    ::MAIN: SickBeard:tv fork auto-detection found custom params {}
2015-12-30 17:02:52 INFO    ::MAIN: SickBeard:tv fork set to custom
2015-12-30 17:02:52 INFO    ::MAIN: FLATTEN: Flattening directory: /mnt/media/downloads/tv/Homeland.S05E12.A.False.Glimmer.1080p.WEB-DL.DD5.1.H.264-NTb
2015-12-30 17:02:52 DEBUG   ::MAIN: Checking for empty folders in:/mnt/media/downloads/tv/Homeland.S05E12.A.False.Glimmer.1080p.WEB-DL.DD5.1.H.264-NTb
2015-12-30 17:02:52 POSTPROCESS::SICKBEARD: SUCCESS: The download succeeded, sending a post-process request
2015-12-30 17:02:52 DEBUG   ::SICKBEARD: Opening URL: http://192.168.188.201:8083/home/postprocess/processEpisode with params: {'nzbName': 'Homeland.S05E12.A.False.Glimmer.1080p.WEB-DL.DD5.1.H.264-NTb', 'quiet': 1}
2015-12-30 17:02:52 POSTPROCESS::SICKBEARD: <html>
2015-12-30 17:02:52 POSTPROCESS::SICKBEARD: <head>
2015-12-30 17:02:52 POSTPROCESS::SICKBEARD:     <title>Mako Runtime Error</title>
2015-12-30 17:02:52 POSTPROCESS::SICKBEARD:     <style>
2015-12-30 17:02:52 POSTPROCESS::SICKBEARD:         body { font-family:verdana; margin:10px 30px 10px 30px;}
2015-12-30 17:02:52 POSTPROCESS::SICKBEARD:         .stacktrace { margin:5px 5px 5px 5px; }
2015-12-30 17:02:52 POSTPROCESS::SICKBEARD:         .highlight { padding:0px 10px 0px 10px; background-color:#9F9FDF; }
2015-12-30 17:02:52 POSTPROCESS::SICKBEARD:         .nonhighlight { padding:0px; background-color:#DFDFDF; }
2015-12-30 17:02:52 POSTPROCESS::SICKBEARD:         .sample { padding:10px; margin:10px 10px 10px 10px;
2015-12-30 17:02:52 POSTPROCESS::SICKBEARD:                   font-family:monospace; }
2015-12-30 17:02:52 POSTPROCESS::SICKBEARD:         .sampleline { padding:0px 10px 0px 10px; }
2015-12-30 17:02:52 POSTPROCESS::SICKBEARD:         .sourceline { margin:5px 5px 10px 5px; font-family:monospace;}
2015-12-30 17:02:52 POSTPROCESS::SICKBEARD:         .location { font-size:80%; }
2015-12-30 17:02:52 POSTPROCESS::SICKBEARD:         .highlight { white-space:pre; }
2015-12-30 17:02:52 POSTPROCESS::SICKBEARD:         .sampleline { white-space:pre; }
2015-12-30 17:02:52 POSTPROCESS::SICKBEARD:     </style>
2015-12-30 17:02:52 POSTPROCESS::SICKBEARD: </head>
2015-12-30 17:02:52 POSTPROCESS::SICKBEARD: <body>
2015-12-30 17:02:52 POSTPROCESS::SICKBEARD: <h2>Error !</h2>
2015-12-30 17:02:52 POSTPROCESS::SICKBEARD: <h3>KeyError: &#39;proc_dir&#39;</h3>
2015-12-30 17:02:52 POSTPROCESS::SICKBEARD:     <div class="sample">
2015-12-30 17:02:52 POSTPROCESS::SICKBEARD:     <div class="nonhighlight">
2015-12-30 17:02:52 POSTPROCESS::SICKBEARD:     
2015-12-30 17:02:52 POSTPROCESS::SICKBEARD:                          if k.lower() not in [&#39;proc_dir&#39;, &#39;dir&#39;, &#39;nzbname&#39;, &#39;process_method&#39;, &#39;proc_type&#39;] else v
2015-12-30 17:02:52 POSTPROCESS::SICKBEARD:     
2015-12-30 17:02:52 POSTPROCESS::SICKBEARD:                          ) for k, v in kwargs.items())
2015-12-30 17:02:52 POSTPROCESS::SICKBEARD:     
2015-12-30 17:02:52 POSTPROCESS::SICKBEARD:         
2015-12-30 17:02:52 POSTPROCESS::SICKBEARD:     
2015-12-30 17:02:52 POSTPROCESS::SICKBEARD:     
2015-12-30 17:02:52 POSTPROCESS::SICKBEARD:                 if not pp_options[b&#34;proc_dir&#34;]:
2015-12-30 17:02:52 POSTPROCESS::SICKBEARD:     
2015-12-30 17:02:52 POSTPROCESS::SICKBEARD:     
2015-12-30 17:02:52 POSTPROCESS::SICKBEARD:                     return self.redirect(&#34;/home/postprocess/&#34;)
2015-12-30 17:02:52 POSTPROCESS::SICKBEARD:     
2015-12-30 17:02:52 POSTPROCESS::SICKBEARD:         
2015-12-30 17:02:52 POSTPROCESS::SICKBEARD:     
2015-12-30 17:02:52 POSTPROCESS::SICKBEARD:                 result = processTV.processDir(pp_options[b&#34;proc_dir&#34;], **pp_options)
2015-12-30 17:02:52 POSTPROCESS::SICKBEARD:     
2015-12-30 17:02:52 POSTPROCESS::SICKBEARD:                 if pp_options.get(&#34;quiet&#34;, None):
2015-12-30 17:02:52 POSTPROCESS::SICKBEARD:     
2015-12-30 17:02:52 POSTPROCESS::SICKBEARD:                     return result
2015-12-30 17:02:52 POSTPROCESS::SICKBEARD:     </div>
2015-12-30 17:02:52 POSTPROCESS::SICKBEARD:     </div>
2015-12-30 17:02:52 POSTPROCESS::SICKBEARD: <div class="stacktrace">
2015-12-30 17:02:52 POSTPROCESS::SICKBEARD:     <div class="location">/usr/local/sickrage/sickbeard/webserve.py, line 2242:</div>
2015-12-30 17:02:52 POSTPROCESS::SICKBEARD:     <div class="nonhighlight">
2015-12-30 17:02:52 POSTPROCESS::SICKBEARD:     
2015-12-30 17:02:52 POSTPROCESS::SICKBEARD:       <div class="sourceline">if not pp_options[b&#34;proc_dir&#34;]:</div>
2015-12-30 17:02:52 POSTPROCESS::SICKBEARD:     </div>
2015-12-30 17:02:52 POSTPROCESS::SICKBEARD:     <div class="location">/usr/local/sickrage/sickbeard/webserve.py, line 221:</div>
2015-12-30 17:02:52 POSTPROCESS::SICKBEARD:     <div class="nonhighlight">
2015-12-30 17:02:52 POSTPROCESS::SICKBEARD:     
2015-12-30 17:02:52 POSTPROCESS::SICKBEARD:       <div class="sourceline">return function(**kwargs)</div>
2015-12-30 17:02:52 POSTPROCESS::SICKBEARD:     </div>
2015-12-30 17:02:52 POSTPROCESS::SICKBEARD: </div>
2015-12-30 17:02:52 POSTPROCESS::SICKBEARD: </body>
2015-12-30 17:02:52 POSTPROCESS::SICKBEARD: </html>
2015-12-30 17:02:52 ERROR   ::MAIN: A problem was reported in the /mnt/media/nzbget/postproc/nzbToSickBeard.py script

Seems like they added a parameter maybe?
